drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -6847,18 +6847,10 @@ Error RenderingDeviceVulkan::_draw_list_render_pass_begin(Framebuffer *framebuff
 	render_pass_begin.pNext = nullptr;
 	render_pass_begin.renderPass = render_pass;
 	render_pass_begin.framebuffer = vkframebuffer;
-	/*
-	 * Given how API works, it makes sense to always fully operate on the whole framebuffer.
-	 * This allows better continue operations for operations like shadowmapping.
 	render_pass_begin.renderArea.extent.width = viewport_size.width;
```

Comment:
So the problem here is that, according to @reduz's experimentation, there is a performance penalty for this on desktop. We need to make this switchable so we only apply this with the Vulkan mobile renderer.

Reply:
"Desktop"? The issue is talking about Mali chips?

Reply:
Yes, that's why separating it and only applying it for mobile builds was suggested? It *helping* with those chips, but has a performance penalty on desktop.

Reply:
Remember that the Mobile rendering method can be used on desktop platforms too, so this may be worth gating based on vendor name rather than the rendering method used.